core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -148,14 +159,20 @@ private ZooKeeper getZooKeeper() {
   }
 
   private class ZCacheWatcher implements Watcher {
+
     @Override
     public void process(WatchedEvent event) {
       if (log.isTraceEnabled()) {
         log.trace("{}", event);
       }
+      setWatcherForTableConfigVersion(event);
```

Comment:
Will an event always trigger this and set it up?  Could this be called in constructor instead?

Reply:
I see what you are saying but unfortunately I don't have way to obtain the Accumulo Instance ID in the ZooCache constructor.   The zoopath of any event that occurs will have the Instance Id in it which I can 'regex out'.   I also am sure that ZooKeeper is connected and functioning by setting the watcher for the 'table-config-version' znode at this point in execution.  

I looked hard for alternatives to doing it this way.  Trust me.